src/basic/hashmap.c

```diff
@@ -1805,6 +1805,19 @@ char** _hashmap_get_strv(HashmapBase *h) {
         return sv;
 }
 
+char** set_to_strv(Set *s) {
+        /* This is similar to set_get_strv(), but steals all contents from the passed Set. */
```

Comment:
Maybe take the ownership of the whole Set? Makes callsites self-explanatory that way